drivers/mtd/mtd.c

```diff
@@ -36,6 +36,11 @@ int mtd_init(mtd_dev_t *mtd)
 
     int res = -ENOTSUP;
 
+    if (mtd->write_size == 0) {
+        DEBUG("mtd: write granularity not set, pessimistically assuming page size\n");
+        mtd->write_size = mtd->page_size;
```

Comment:
Now this is terrible.
So far everything assumed a write granularity of 1 and it worked.
This would break *every* MTD backed or at least incur a costly work-around in the MTD layer for *every* driver.

If you really want to avoid setting an optimistic default, just set a default for each MTD driver - there are not *that* many :wink:

Reply:
Setting a default for each MTD is certainly an option -- but at least for one I looked at the data sheet and it wasn't very explicit about any requirements. That probably indicates it has a granularity of 1 (because for a managed device that'd be typical). Do you know them well enough to make a general statement on everything but flashpage_mtd that we have in-tree to say that 1 is the value for them? (Or at least for some? Then I'd set *these* to 1 already in this PR.)

(And about "it worked", that's likely just because it hasn't been tested on flashpage_mtd on all devices that have flashpage.)

Reply:
The EEPROMs (at24, at25) certainly can write individual bytes. Same for SPI NOR flash.
The SD card driver only supports blocks of 512 bytes unless `mtd_write_page` is used (#17619). 
`mtd_mapper` should just inherit the value of the underlying MTD device.